File: src/main/java/net/frozenorb/potpvp/util/nametag/NametagThread.java
package net.frozenorb.potpvp.util.nametag;

import lombok.Getter;
import net.frozenorb.potpvp.PotPvPRP;
import net.frozenorb.potpvp.util.nametag.construct.NameTagUpdate;

import java.util.Iterator;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

@Getter
public class NametagThread extends Thread {

    private final Map<NameTagUpdate, Boolean> pendingUpdates = new ConcurrentHashMap<>();

    public NametagThread() {
        super("PotPvPSI - NameTags Thread");
        setDaemon(false);
    }

    public void run() {
        while (true) {
            Iterator<NameTagUpdate> pendingUpdatesIterator = pendingUpdates.keySet().iterator();

            while(pendingUpdatesIterator.hasNext()) {
                NameTagUpdate pendingUpdate = pendingUpdatesIterator.next();

                try {
                    PotPvPRP.getInstance().getNameTagHandler().applyUpdate(pendingUpdate);
                    pendingUpdatesIterator.remove();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }

            try {
                Thread.sleep(2 * 50L);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }

}

File: src/main/kotlin/net/frozenorb/potpvp/kt/nametag/NametagThread.kt
package net.frozenorb.potpvp.kt.nametag

import net.frozenorb.potpvp.PotPvPRP
import java.util.concurrent.ConcurrentHashMap

internal class NametagThread : Thread("stark - Nametag Thread") {
    init {
        this.isDaemon = true
    }

    override fun run() {
        while (true) {
            val pendingUpdatesIterator = pendingUpdates.keys.iterator()
            while (pendingUpdatesIterator.hasNext()) {
                val pendingUpdate = pendingUpdatesIterator.next()
                try {
                    PotPvPRP.getInstance().nameTagHandler.applyUpdate(pendingUpdate)
                    pendingUpdatesIterator.remove()
                } catch (e: Exception) {
                    e.printStackTrace()
                }

            }

            try {
                sleep(PotPvPRP.getInstance().nameTagHandler.updateInterval * 50L)
            } catch (e2: InterruptedException) {
                e2.printStackTrace()
            }

        }
    }

    companion object {
        var pendingUpdates = ConcurrentHashMap<NametagUpdate, Boolean>()
    }
}